### PR TITLE
Feature/alarm

### PR DESCRIPTION
**Pull request by cutieskye**

### CODE REVIEW

**Comment by cutieskye** [comment]
https://user-images.githubusercontent.com/68032402/211614248-1a7cda56-bb2c-4bf2-9fff-e77fcf1a94cb.mp4

**Comment by simonmicro** [comment]
I've noted some similarities with the other PRs. Could you please leave a note in them, which PR is based upon what? This would save me some time by not reviewing your changes twice (and commenting them :P ).

File: emulator/src/tests/OswServiceTaskNotifier.cpp [hunk 1]
```diff
@@ -1,123 +0,0 @@
-#include "utest.h"
```
Comment (simonmicro) [inline emulator/src/tests/OswServiceTaskNotifier.cpp:1]: Why did you remove the unit test?

**Review by simonmicro — COMMENTED** [review]
...and when should this be merged? After or before #343 ?

**Comment by akmal-threepointsix** [comment]
> ...and when should this be merged? After or before #343 ?

The order does not matter, since they are independent features. 

And with any order there will be few merge conflicts 😃, but they are very easy to fix 😉.

File: emulator/src/tests/OswServiceTaskNotifier.cpp (hunk 1 above)
Comment (simonmicro) [reply emulator/src/tests/OswServiceTaskNotifier.cpp:1]: @cutieskye  @akmal-threepointsix

**Review by simonmicro — APPROVED** [review]
Well, while the comment inside the unit test for the `OswServiceTaskNotifier` may not be true anymore, I'm not a fan of implementing it myself... @cutieskye @akmal-threepointsix ?

**Comment by simonmicro** [comment]
A note regarding the "fixtures": A fixture is something which prepares / resets something for one specific unit-test. In the other tests those are used to start/prepare the emulator if e.g. interaction with the HAL is needed or used to reset/prepare the Preferences/NVS before every test run. Your unit-tests are therefore not really well placed inside the fixtures-folder. Furthermore, a fixture shall be independent of the test, so others could potentially reuse them as needed.

**Comment by akmal-threepointsix** [comment]
I resolved all conflicts.

> Well, while the comment inside the unit test for the `OswServiceTaskNotifier` may not be true anymore, I'm not a fan of implementing it myself... @cutieskye @akmal-threepointsix ?

Yes, I can fix that. But should I do it before or after the merge?

**Review by RuffaloLavoisier — APPROVED** [review]
LGTM - We don't have a trigger module at the moment, but the idea is great!

**Comment by akmal-threepointsix** [comment]
Hmm, one check failed. But it looks like it is not our fault, it just could not download some package 🤔. Could someone restart all checks?

**Comment by simonmicro** [comment]
> Hmm, one check failed. But it looks like it is not our fault, it just could not download some package thinking. Could someone restart all checks?

Consider it done :)

**Comment by simonmicro** [comment]
I'm sorry, but your branch is somewhat out-of-sync with the `develop`, so I wasn't able to merge it. I've gone ahead and merged #350 before yours (because of your conflicts). Would it be possible to that you rebase on top of the current `develop` branch? I'll try to merge yours ASAP :sweat:

**Comment by akmal-threepointsix** [comment]
Erm sorry, I accidentally merged "develop" into this branch instead of rebasing. Is it ok?

**Comment by simonmicro** [comment]
Nope, the "Merge Queue" is still rejecting this branch... If you can't solve it I would swing my admin-mode and hammer it in :)
